client/ayon_core/tools/utils/widgets.py

```diff
@@ -104,6 +105,151 @@ def __init__(self, *args, **kwargs):
+            w_size_policy.setVerticalPolicy(
+                QtWidgets.QSizePolicy.MinimumExpanding)
+            widget.setSizePolicy(w_size_policy)
+
+        # Set size hint of this frame to fixed so size hint height is
+        #   used as fixed height
+        size_policy = self.sizePolicy()
+        size_policy.setVerticalPolicy(QtWidgets.QSizePolicy.Fixed)
+        self.setSizePolicy(size_policy)
+
+        text_input.returnPressed.connect(self.returnPressed)
+        text_input.textChanged.connect(self.textChanged)
+        text_input.textEdited.connect(self.textEdited)
+        options_button.clicked.connect(self._on_options_button_clicked)
+
+        self._text_input = text_input
+        self._options_button = options_button
+        self._options = None
+
+        # Set default state
+        self.set_options(options)
+
+    def text(self) -> str:
+        return self._text_input.text()
+
+    def setText(self, text: str):
+        self._text_input.setText(text)
+
+    def setPlaceholderText(self, text: str):
+        self._text_input.setPlaceholderText(text)
+
+    def placeholderText(self) -> str:
+        return self._text_input.placeholderText()
+
+    def setReadOnly(self, state: bool):
+        self._text_input.setReadOnly(state)
+
+    def setIcon(self, icon: QtGui.QIcon):
+        self._options_button.setIcon(icon)
+
+    def setToolTip(self, text: str):
+        self._text_input.setToolTip(text)
+
+    def set_button_tool_tip(self, text: str):
+        self._options_button.setToolTip(text)
+
+    def set_options(self, options: Optional[List[str]] = None):
+        self._options = options
+        self._options_button.setEnabled(bool(options))
+
+    def sizeHint(self) -> QtCore.QSize:
+        hint = super().sizeHint()
+        tsz = self._text_input.sizeHint()
+        bsz = self._options_button.sizeHint()
+        hint.setHeight(max(tsz.height(), bsz.height()))
+        return hint
+
+    # Adds ability to change style of the widgets
+    # - because style change of the 'HintedLineEdit' may not propagate
+    #   correctly 'HintedLineEditInput' and 'HintedLineEditButton'
+    def set_text_widget_object_name(self, name: str):
+        self._text_input.setObjectName(name)
+
+    def set_text_widget_property(self, name: str, value: Any):
+        set_style_property(self._text_input, name, value)
+
+    def set_button_widget_object_name(self, name: str):
+        self._text_input.setObjectName(name)
+
+    def set_button_widget_property(self, name: str, value: Any):
+        set_style_property(self._options_button, name, value)
+
+    def _on_options_button_clicked(self):
+        if not self._options:
+            return
+
+        menu = QtWidgets.QMenu(self)
+        for option in self._options:
+            if option in self.SEPARATORS:
+                menu.addSeparator()
+                continue
+            action = menu.addAction(option)
+            action.triggered.connect(self._on_option_action)
```

Comment:
In this case - since you're responding to the action any way, you can also add the callback to the menu instead so we don't need to per action, and then in the callback respond to the action.

```python
menu.triggered.connect(self._on_option_action)

def on_option_action(self, action):
    self.setText(action.text())
```
Something like that?

Reply:
Didn't know could be done, will do...